### ISSUE TITLE
Depository compatibility issues

### ISSUE THREAD

**Issue by Ashoka74** [MagGeo/MagGeo-Annotation-Program#1]
While installing the package, I figured that I went through some difficulties where I had to :

1) Reinstall some modules (Numpy, Scipy, Matplotlib..) with pip instead of conda
2) Specify full path to modules within my jupyter notebook & MagGeoFunctions.py file (mostly struggled with chaosmagpy & cartopy + dependencies)
3) Bypass SSL_Verifications (some issues with Certifi) : 
import ssl
ssl._create_default_https_context = ssl._create_unverified_context
4) Could not use geoplot to plot on a map at the end because modules Rasterio, GDAL & Python had some incompatibilities

I might have faced these issues because I made a mistake during the environment setting (maybe some packages were already installed through pip & created incompatibility).

This is the distribution of my modules: 

conda list --show-channel-urls

#
# Name                    Version                   Build  Channel
argon2-cffi               20.1.0           py38h2bbff1b_1    defaults
async_generator           1.10               pyhd3eb1b0_0    defaults
attrs                     21.2.0             pyhd3eb1b0_0    defaults
backcall                  0.2.0              pyhd3eb1b0_0    defaults
beautifulsoup4            4.10.0                   pypi_0    pypi
blas                      1.0                         mkl    defaults
bleach                    4.0.0              pyhd3eb1b0_0    defaults
blosc                     1.21.0               h0e60522_0    conda-forge
bokeh                     2.3.3            py38haa95532_0    defaults
boost-cpp                 1.74.0               h5b4e17d_4    conda-forge
bottleneck                1.3.2            py38h2a96729_1    defaults
brotlipy                  0.7.0           py38h2bbff1b_1003    defaults
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2021.9.30            haa95532_1    defaults
cached-property           1.5.2                      py_0    defaults
cairo                     1.16.0            hb19e0ff_1008    conda-forge
cdflib                    0.3.20             pyhd8ed1ab_0    conda-forge
certifi                   2021.10.8        py38haa95532_0    defaults
cffi                      1.14.6           py38h2bbff1b_0    defaults
cfitsio                   3.470                he774522_6    defaults
cftime                    1.5.1                    pypi_0    pypi
chaosmagpy                0.7.1                    pypi_0    pypi
chardet                   4.0.0                    pypi_0    pypi
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.0.3                    pypi_0    pypi
colorama                  0.4.4              pyhd3eb1b0_0    defaults
cryptography              3.4.8            py38h71e12ea_0    defaults
curl                      7.79.1               h789b8ee_1    conda-forge
cython                    0.29.24          py38hd77b12b_0    defaults
debugpy                   1.4.1            py38hd77b12b_0    defaults
decorator                 5.1.0              pyhd3eb1b0_0    defaults
defusedxml                0.7.1              pyhd3eb1b0_0    defaults
dnspython                 2.1.0                    pypi_0    pypi
entrypoints               0.3                      py38_0    defaults
expat                     2.4.1                h6c2663c_2    defaults
font-ttf-dejavu-sans-mono 2.37                 hd3eb1b0_0    defaults
font-ttf-inconsolata      2.001                hcb22688_0    defaults
font-ttf-source-code-pro  2.030                hd3eb1b0_0    defaults
font-ttf-ubuntu           0.83                 h8b1ccd4_0    defaults
fontconfig                2.13.1            h1989441_1005    conda-forge
fonts-anaconda            1                    h8fa9717_0    defaults
fonts-conda-ecosystem     1                    hd3eb1b0_0    defaults
freetype                  2.10.4               hd328e21_0    defaults
freexl                    1.0.6                h2bbff1b_0    defaults
gdal                      3.3.2            py38hc3a09ef_4    conda-forge
geojson                   2.5.0                      py_0    conda-forge
geos                      0.2.3                    pypi_0    pypi
geotiff                   1.7.0                h350af67_3    conda-forge
gettext                   0.19.8.1          ha2e2712_1008    conda-forge
h5py                      3.4.0           nompi_py38hde0384b_101    conda-forge
hdf4                      4.2.15               h0e5069d_3    conda-forge
hdf5                      1.12.1          nompi_h2a0e4a3_101    conda-forge
hdf5storage               0.1.18                   pypi_0    pypi
icc_rt                    2019.0.0             h0cc432a_1    defaults
icu                       68.1                 h6c2663c_0    defaults
idna                      3.2                pyhd3eb1b0_0    defaults
importlib-metadata        4.8.1            py38haa95532_0    defaults
importlib_metadata        4.8.1                hd3eb1b0_0    defaults
intel-openmp              2021.3.0          haa95532_3372    defaults
ipykernel                 6.4.1            py38haa95532_1    defaults
ipython                   7.27.0           py38hd4e2768_0    defaults
ipython_genutils          0.2.0              pyhd3eb1b0_1    defaults
jbig                      2.1               h8d14728_2003    conda-forge
jedi                      0.18.0           py38haa95532_1    defaults
jinja2                    2.11.3                   pypi_0    pypi
joblib                    1.1.0                    pypi_0    pypi
jpeg                      9d                   h2bbff1b_0    defaults
jsonschema                3.2.0              pyhd3eb1b0_2    defaults
jupyter_client            7.0.1              pyhd3eb1b0_0    defaults
jupyter_core              4.8.1            py38haa95532_0    defaults
jupyterlab_pygments       0.1.2                      py_0    defaults
kealib                    1.4.14               h8995ca9_3    conda-forge
krb5                      1.19.2               h5b6d351_0    defaults
lcms2                     2.12                 h83e58a3_0    defaults
lerc                      3.0                  h0e60522_0    conda-forge
libcurl                   7.79.1               h789b8ee_1    conda-forge
libdeflate                1.8                  h2bbff1b_5    defaults
libffi                    3.4.2                hd77b12b_0    defaults
libgdal                   3.3.2                h08edb8f_4    conda-forge
libglib                   2.70.0               h3be07f2_0    conda-forge
libiconv                  1.16                 he774522_0    conda-forge
libkml                    1.3.0             h9859afa_1014    conda-forge
libnetcdf                 4.8.1           nompi_h1cc8e9d_101    conda-forge
libpng                    1.6.37               h2a8f88b_0    defaults
libpq                     13.3                 hfcc5ef8_1    conda-forge
librttopo                 1.1.0                hb340de5_6    conda-forge
libspatialite             5.0.1                h6a90f17_9    conda-forge
libssh2                   1.10.0               h680486a_2    conda-forge
libtiff                   4.3.0                hd413186_2    conda-forge
libwebp                   1.2.0                h2bbff1b_0    defaults
libwebp-base              1.2.0                h2bbff1b_0    defaults
libxml2                   2.9.12               h0ad7f3c_0    defaults
libzip                    1.8.0                hfed4ece_1    conda-forge
libzlib                   1.2.11            h8ffe710_1013    conda-forge
linkchecker               10.0.1                   pypi_0    pypi
lxml                      4.6.3                    pypi_0    pypi
lz4-c                     1.9.3                h2bbff1b_1    defaults
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.0.1            py38h2bbff1b_0    defaults
matplotlib-inline         0.1.2              pyhd3eb1b0_2    defaults
mistune                   0.8.4           py38he774522_1000    defaults
mkl                       2021.3.0           haa95532_524    defaults
mkl-service               2.4.0            py38h2bbff1b_0    defaults
mkl_fft                   1.3.1            py38h277e83a_0    defaults
mkl_random                1.2.2            py38hf11a4ad_0    defaults
mock                      4.0.3              pyhd3eb1b0_0    defaults
msys2-conda-epoch         20160418                      1    conda-forge
nbclient                  0.5.3              pyhd3eb1b0_0    defaults
nbconvert                 6.1.0            py38haa95532_0    defaults
nbformat                  5.1.3              pyhd3eb1b0_0    defaults
nest-asyncio              1.5.1              pyhd3eb1b0_0    defaults
netcdf4                   1.5.7                    pypi_0    pypi
notebook                  6.4.3            py38haa95532_0    defaults
numexpr                   2.7.3            py38hb80d3ca_1    defaults
numpy                     1.21.3                   pypi_0    pypi
olefile                   0.46               pyhd3eb1b0_0    defaults
openjpeg                  2.4.0                h4fc8c34_0    defaults
openssl                   1.1.1l               h2bbff1b_0    defaults
packaging                 21.0               pyhd3eb1b0_0    defaults
pandas                    1.3.3            py38h6214cd6_0    defaults
pandocfilters             1.4.3            py38haa95532_1    defaults
parso                     0.8.2              pyhd3eb1b0_0    defaults
pcre                      8.45                 hd77b12b_0    defaults
pickleshare               0.7.5           pyhd3eb1b0_1003    defaults
pillow                    8.4.0            py38hd45dc43_0    defaults
pip                       21.0.1           py38haa95532_0    defaults
pixman                    0.40.0               h2bbff1b_1    defaults
poppler                   21.09.0              h24fffdf_3    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                13.3                 h1c22c4f_1    conda-forge
proj                      8.1.1                h1cfcee9_2    conda-forge
prometheus_client         0.11.0             pyhd3eb1b0_0    defaults
prompt-toolkit            3.0.20             pyhd3eb1b0_0    defaults
pycparser                 2.20                       py_2    defaults
pygeos                    0.10.2                   pypi_0    pypi
pygments                  2.10.0             pyhd3eb1b0_0    defaults
pyopenssl                 20.0.1             pyhd3eb1b0_1    defaults
pyparsing                 2.4.7              pyhd3eb1b0_0    defaults
pyrsistent                0.17.3           py38he774522_0    defaults
pysocks                   1.7.1            py38haa95532_0    defaults
pytables                  3.6.1            py38h7deab4d_4    conda-forge
python                    3.8.12               h6244533_0    defaults
python-dateutil           2.8.2              pyhd3eb1b0_0    defaults
python_abi                3.8                      2_cp38    conda-forge
pytz                      2021.3             pyhd3eb1b0_0    defaults
pywin32                   228              py38hbaba5e8_1    defaults
pywinpty                  0.5.7                    py38_0    defaults
pyxdg                     0.27                     pypi_0    pypi
pyyaml                    5.4.1            py38h2bbff1b_1    defaults
pyzmq                     22.2.1           py38hd77b12b_1    defaults
requests                  2.26.0             pyhd3eb1b0_0    defaults
scikit-learn              1.0.1                    pypi_0    pypi
scipy                     1.7.1            py38hbe87c03_2    defaults
send2trash                1.8.0              pyhd3eb1b0_1    defaults
setuptools                58.0.4           py38haa95532_0    defaults
six                       1.16.0             pyhd3eb1b0_0    defaults
sklearn                   0.0                      pypi_0    pypi
soupsieve                 2.2.1                    pypi_0    pypi
sqlite                    3.36.0               h2bbff1b_0    defaults
terminado                 0.9.4            py38haa95532_0    defaults
testpath                  0.5.0              pyhd3eb1b0_0    defaults
threadpoolctl             3.0.0                    pypi_0    pypi
tiledb                    2.3.4                h78dabda_0    conda-forge
tk                        8.6.11               h2bbff1b_0    defaults
tornado                   6.1              py38h2bbff1b_0    defaults
tqdm                      4.62.2             pyhd3eb1b0_1    defaults
traitlets                 5.1.0              pyhd3eb1b0_0    defaults
typing-extensions         3.10.0.2             hd3eb1b0_0    defaults
typing_extensions         3.10.0.2           pyh06a4308_0    defaults
urllib3                   1.26.7                   pypi_0    pypi
vc                        14.2                 h21ff451_1    defaults
viresclient               0.9.1                    pypi_0    pypi
vs2015_runtime            14.27.29016          h5e58377_2    defaults
wcwidth                   0.2.5              pyhd3eb1b0_0    defaults
webencodings              0.5.1                    py38_1    defaults
werkzeug                  2.0.2                    pypi_0    pypi
wheel                     0.37.0             pyhd3eb1b0_1    defaults
win_inet_pton             1.1.0            py38haa95532_0    defaults
wincertstore              0.2              py38haa95532_2    defaults
winpty                    0.4.3                         4    defaults
xarray                    0.19.0             pyhd3eb1b0_1    defaults
xerces-c                  3.2.3                ha925a31_0    defaults
xz                        5.2.5                h62dcd97_0    defaults
yaml                      0.2.5                he774522_0    defaults
zipp                      3.6.0              pyhd3eb1b0_0    defaults
zlib                      1.2.11            h8ffe710_1013    conda-forge
zstd                      1.5.0                h6255e5f_0    conda-forge

**Comment by mfbenitezp** [comment]
Thanks @smithara, @Ashoka74 this has been addressed in https://github.com/MagGeo/MagGeo-Annotation-Program/tree/minconda_version after some testing I will merge into the master.
